zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/RoleRemoveEntityProcessor.java

```diff
@@ -0,0 +1,119 @@
+import io.camunda.zeebe.protocol.record.value.EntityType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+import io.camunda.zeebe.stream.api.state.KeyGenerator;
+
+public class RoleRemoveEntityProcessor implements DistributedTypedRecordProcessor<RoleRecord> {
+
+  private final RoleState roleState;
+  private final UserState userState;
+  private final AuthorizationCheckBehavior authCheckBehavior;
+  private final KeyGenerator keyGenerator;
+  private final StateWriter stateWriter;
+  private final TypedRejectionWriter rejectionWriter;
+  private final TypedResponseWriter responseWriter;
+  private final CommandDistributionBehavior commandDistributionBehavior;
+
+  public RoleRemoveEntityProcessor(
+      final RoleState roleState,
+      final UserState userState,
+      final AuthorizationCheckBehavior authCheckBehavior,
+      final KeyGenerator keyGenerator,
+      final Writers writers,
+      final CommandDistributionBehavior commandDistributionBehavior) {
+    this.roleState = roleState;
+    this.userState = userState;
+    this.authCheckBehavior = authCheckBehavior;
+    this.keyGenerator = keyGenerator;
+    stateWriter = writers.state();
+    rejectionWriter = writers.rejection();
+    responseWriter = writers.response();
+    this.commandDistributionBehavior = commandDistributionBehavior;
+  }
+
+  @Override
+  public void processNewCommand(final TypedRecord<RoleRecord> command) {
+    final var record = command.getValue();
+    final var persistedRecord = roleState.getRole(record.getRoleKey());
+    if (persistedRecord.isEmpty()) {
+      final var errorMessage =
+          "Expected to update role with key '%s', but a role with this key does not exists"
+              .formatted(record.getRoleKey());
+      rejectionWriter.appendRejection(command, RejectionType.NOT_FOUND, errorMessage);
+      responseWriter.writeRejectionOnCommand(command, RejectionType.NOT_FOUND, errorMessage);
+      return;
+    }
+
+    final var authorizationRequest =
+        new AuthorizationRequest(command, AuthorizationResourceType.ROLE, PermissionType.UPDATE)
+            .addResourceId(record.getName());
+    if (!authCheckBehavior.isAuthorized(authorizationRequest)) {
+      final var errorMessage =
+          UNAUTHORIZED_ERROR_MESSAGE.formatted(
+              authorizationRequest.getPermissionType(), authorizationRequest.getResourceType());
+      rejectionWriter.appendRejection(command, RejectionType.UNAUTHORIZED, errorMessage);
+      responseWriter.writeRejectionOnCommand(command, RejectionType.UNAUTHORIZED, errorMessage);
+      return;
+    }
+
+    final var entityKey = record.getEntityKey();
+    final var entityType = record.getEntityType();
+    if (!isEntityPresent(entityKey, entityType)) {
+      final var errorMessage =
+          "Expected to remove entity with key '%s' and type '%s' to role with key '%s', the entity does not exists"
```

Comment:
🙃  Some small suggestions:

```suggestion
          "Expected to remove an entity with key '%s' and type '%s' from role with key '%s', but the entity doesn't exist"
```